WordPress/WordPressAllTimeWidget/AllTimeViewController.swift

```diff
@@ -1,30 +1,183 @@
+
+    // MARK: - Site Configuration
+
+    func retrieveSiteConfiguration() {
+        guard let sharedDefaults = UserDefaults(suiteName: WPAppGroupName) else {
+            DDLogError("All Time Widget: Unable to get sharedDefaults.")
+            isConfigured = false
+            return
+        }
+
+        siteID = sharedDefaults.object(forKey: WPStatsTodayWidgetUserDefaultsSiteIdKey) as? NSNumber
+        siteUrl = sharedDefaults.string(forKey: WPStatsTodayWidgetUserDefaultsSiteUrlKey) ?? Constants.noDataLabel
+        oauthToken = fetchOAuthBearerToken()
+
+        if let timeZoneName = sharedDefaults.string(forKey: WPStatsTodayWidgetUserDefaultsSiteTimeZoneKey) {
+            timeZone = TimeZone(identifier: timeZoneName)
+        }
+
+        isConfigured = siteID != nil && timeZone != nil && oauthToken != nil
+    }
+
+    func fetchOAuthBearerToken() -> String? {
+        let oauth2Token = try? SFHFKeychainUtils.getPasswordForUsername(WPStatsTodayWidgetKeychainTokenKey, andServiceName: WPStatsTodayWidgetKeychainServiceName, accessGroup: WPAppKeychainAccessGroup)
+
+        return oauth2Token as String?
+    }
+
+    // MARK: - Data Management
+
+    func loadSavedData() {
+        statsValues = AllTimeWidgetStats.loadSavedData()
+    }
+
+    func saveData() {
+        statsValues?.saveData()
+    }
+
+    func fetchData() {
+        guard let statsRemote = statsRemote() else {
+            return
+        }
+
+        statsRemote.getInsight { (allTimesStats: StatsAllTimesInsight?, error) in
+            if error != nil {
+                DDLogError("All Time Widget: Error fetching StatsAllTimesInsight: \(String(describing: error?.localizedDescription))")
+                return
+            }
+
+            DDLogDebug("All Time Widget: Fetched StatsAllTimesInsight data.")
+
+            DispatchQueue.main.async {
+                self.statsValues = AllTimeWidgetStats(views: allTimesStats?.viewsCount ?? 0,
+                                            visitors: allTimesStats?.visitorsCount ?? 0,
+                                            posts: allTimesStats?.postsCount ?? 0,
+                                            bestViews: allTimesStats?.bestViewsPerDayCount ?? 0)
+
+                // TODO: reload table here
+            }
+        }
+    }
+
+    func statsRemote() -> StatsServiceRemoteV2? {
+        guard
+            let siteID = siteID,
+            let timeZone = timeZone,
+            let oauthToken = oauthToken
+            else {
+                DDLogError("All Time Widget: Missing site ID, timeZone or oauth2Token")
+                return nil
+        }
+
+        let wpApi = WordPressComRestApi(oAuthToken: oauthToken)
+        return StatsServiceRemoteV2(wordPressComRestApi: wpApi, siteID: siteID.intValue, siteTimezone: timeZone)
+    }
+
+    // MARK: - Helpers
+
+    func displayString(for value: Int) -> String {
+        return numberFormatter.string(from: NSNumber(value: value)) ?? "0"
```

Comment:
Should we default to `noDataLabel` instead of 0 if we can't format a number? It just might better indicate that we couldn't load the value, rather than that there's actually data with a value of 0 there.

Reply:
I elected to show the unformatted value instead. I think that's better than the `noDataLabel`.